plus/src/main/java/org/apache/calcite/adapter/os/OsQueryTableUtil.java

```diff
@@ -0,0 +1,386 @@
+      case 3:
+        break;
+      // TYPE_RAM_DISK ：Flash memory
+      case 4:
+        break;
+      // TYPE_CDROM ：Optical drive
+      case 5:
+        break;
+      // TYPE_SWAP ：Page exchang
+      case 6:
+        break;
+      default:
+        break;
+      }
+      list.add(objects);
+    }
+    return list;
+  }
+
+  private List<Object[]> whoUse() throws SigarException {
+    final List<Object[]> list = new ArrayList<Object[]>();
+    final Who[] whos = sigar.getWhoList();
+    if (whos != null && whos.length > 0) {
+      for (Who who : whos) {
+        Object[] objects = new Object[4];
+        objects[0] = who.getDevice();
+        objects[1] = who.getHost();
+        objects[2] = who.getUser();
+        objects[3] = who.getTime();
+        list.add(objects);
+      }
+    }
+    return list;
+  }
+
+  private void loadLibs() {
+    if (StringUtils.isEmpty(System.getProperty("org.hyperic.sigar.path"))) {
+      String sigarPathStr = "/hyperic-sigar-1.6.4/sigar-bin/lib/";
+      String prop = System.getProperty("org.hyperic.sigar.path", sigarPathStr);
+      URL sigarPathUrl = OsQueryTableUtil.class.getResource(prop);
+      System.setProperty("org.hyperic.sigar.path", Sources.of(sigarPathUrl).file().getPath());
+    }
+  }
+
+  public List<Object[]> getResultInfo() {
+    this.sigar = new Sigar();
+    List<Object[]> objs = new ArrayList<Object[]>();
+    try {
+      switch (type) {
+      case SYSTEM_INFO:
+        objs = systemInfo();
+        break;
+      case JAVA_INFO:
+        objs = javaInfo();
+        break;
+      case OS_VERSION:
+        objs = osInfo();
+        break;
+      case MEMORY_INFO:
+        objs = memoryInfo();
+        break;
+      case CPU_INFO:
+        objs = cpuInfo();
+        break;
+      case INTERFACE_DETAILS:
+        objs = interfaceDetails();
+        break;
+      case INTERFACE_ADDRESSES:
+        objs = interfaceAddresses();
+        break;
+      case MOUNTS:
+        objs = mounts();
+        break;
+      case WHO_USE:
+        objs = whoUse();
+        break;
+      default:
+        break;
+      }
+    } catch (Exception e1) {
+      e1.printStackTrace();
+    }
+    return objs;
+  }
+
+  /**
+   * Storage unit conversion tool.
+   */
+  public static class HexTransformationUtils {
+    public static Integer byteToTb(Long number) {
+      Long standard = 1024 * 1024 * 1024 * 1024L;
+      Long calculation = (long) (number / standard);
+      Integer result = calculation.intValue();
+      return result;
+    }
+
+    public static Integer gbToTb(Integer number) {
+      Integer standard = 1024;
+      Integer result = (int) (number / standard);
+      return result;
+    }
+
+    public static Integer kbToGb(Long number) {
+      Integer standard = 1024 * 1024;
+      Integer result = (int) (number / standard);
+      return result;
+    }
+  }
+
+  /**
+   * Get system enumeration information.
+   */
+  public enum OsQueryEnum {
+    SYSTEM_INFO,
```

Comment:
It seems that this enumeration is only for internal use, we can make it package scope.